src/modules/NodeCollection.js

```diff
@@ -0,0 +1,77 @@
+export class NodeCollection {
+  constructor (nodes) {
+    this.nodes = nodes
+  }
+
+  _strValue (str, caseSensitive) {
+    return caseSensitive ? `${str}` : `${str}`.toLowerCase()
+  }
+
+  _createStringFilter (filterValue, caseSensitive) {
+    if (filterValue === undefined || filterValue === null) {
+      filterValue = ''
+    }
+    const strFilter = this._strValue(filterValue, caseSensitive)
+    return value => this._strValue(value, caseSensitive).indexOf(strFilter) >= 0
+  }
+
+  _filterByProps (node, properties, filter) {
+    const mergedProps = [properties].reduce(
+      (merged, prop) => merged.concat(prop),
+      []
+    )
+    return mergedProps.find(prop => filter(node[prop]))
+  }
+
+  filter (properties, filter) {
+    const filtered = this.nodes.filter(node =>
+      this._filterByProps(node, properties, filter)
+    )
+    return new NodeCollection(filtered)
+  }
+
+  contains (properties, value, caseSensitive = false) {
+    return this.filter(
+      properties,
+      this._createStringFilter(value, caseSensitive)
+    )
+  }
+
+  equals (properties, value) {
+    return this.filter(properties, nodeValue => value === nodeValue)
+  }
+
+  betweenNumber (properties, minValue, maxValue) {
+    return this.filter(properties, nodeValue =>
+      (minValue === undefined || minValue === null || minValue === '' || minValue <= nodeValue) &&
+      (maxValue === undefined || maxValue === null || maxValue === '' || maxValue >= nodeValue)
+    )
+  }
+
+  betweenDate (properties, minValue, maxValue) {
+    return this.filter(properties, nodeValue =>
+      (minValue === undefined || minValue === null || minValue === '' || new Date(minValue) <= nodeValue) &&
```

Comment:
Isn't it better to use `.time()` and use timestamps to compare dates?

Reply:
Yes, will change it.

Reply:
Also I would create a method for this like `isUndefined(value) => return value === undefined || value === null || value === ''

Reply:
You are absolutely right, I also wanted to get rid of the repetitive and verbose undefined checking :-).

Reply:
This is done.